components/tabs/src/storage.rs

```diff
@@ -72,15 +78,26 @@ pub struct ClientRemoteTabs {
 pub struct TabsStorage {
     local_tabs: RefCell<Option<Vec<RemoteTab>>>,
     db_path: PathBuf,
-    db_connection: Option<Connection>,
+    db_connection: DbConnection,
 }
 
 impl TabsStorage {
     pub fn new(db_path: impl AsRef<Path>) -> Self {
         Self {
             local_tabs: RefCell::default(),
             db_path: db_path.as_ref().to_path_buf(),
-            db_connection: None,
+            db_connection: DbConnection::Created,
+        }
+    }
+
+    pub fn close(&mut self) -> Result<bool> {
+        match std::mem::replace(&mut self.db_connection, DbConnection::Closed) {
+            DbConnection::Open(conn) => {
+                // Now we own the connection, so we can close it.
+                conn.close().map_err(|(_conn, err)| err)?;
```

Comment:
Thought: I wonder if it's worth swallowing `close()` errors, and having this method return nothing. On the one hand, it does lose context if closing fails; on the other...I'm not super sure how a caller _would_ recover from a `close()` failure.

Thinking a bit more about it, if `close()` fails inside an `AsyncShutdown` blocker, it's just going to print a message to the Firefox JS console, and continue with shutdown, so I wonder if we can simplify that even more, and make this non-throwing.

WDYT?

Reply:
I'm inclined to agree with you here! I decided to keep the error logged just so anyone potentially running into issues can see what happened but agree that bubbling the error doesn't really help in this scenario. Certainly simplifies the code!